### ISSUE TITLE
terraform plan -refresh=true -destroy does not refresh state (0.14.X)

### ISSUE THREAD

**Issue by circa10a** [hashicorp/terraform#28847]
### Terraform Version

```
0.14.11
```

### Terraform Configuration Files

```terraform
resource "local_file" "foo" {
    content     = "foo!"
    filename = "${path.module}/foo.bar"
}
```

### Expected Behavior

TLDR; With terraform v0.14.x does not refresh state during terraform plans when `-destroy` is passed, but does when performing a normal plan.

After upgrading to terraform 0.14.11 from 0.13.6, we've noticed several terraform destroy's error with things like launch configuration not found, but after a `terraform refresh` then running terraform destroy again, terraform is able to find the launch config(no manual changes in the console). This is an expensive issue because we now have lots of rogue resources running in our cloud infrastructure.

Upon further inspection, it seems `terraform plan -destroy` does not actually refresh state (but does in terraform `0.13.6` and `0.15.4`.  Even when running `terraform plan -refresh=true -destroy`,  doesn't refresh

**Terraform v0.13.6**

```
❯ ~/bin/terraform plan -refresh=true -destroy
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

local_file.foo: Refreshing state... [id=4bf3e335199107182c6f7638efaad377acc7f452]
```

**Terraform v0.15.4**

```
❯ terraform plan -destroy 
local_file.foo: Refreshing state... [id=4bf3e335199107182c6f7638efaad377acc7f452]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  - destroy
```

I expect terraform to refresh state during plan with `-destroy`

### Actual Behavior

No state was refreshed when running `terraform plan -destroy`

Terraform v0.14.0 && v0.14.11

```
❯ ~/bin/terraform plan -refresh=true -destroy

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # local_file.foo will be destroyed
  - resource "local_file" "foo" {
      - content              = "foo!" -> null
      - directory_permission = "0777" -> null
      - file_permission      = "0777" -> null
      - filename             = "./foo.bar" -> null
      - id                   = "4bf3e335199107182c6f7638efaad377acc7f452" -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.
```

 **State does not refresh, but normal plan does**
 
 ```
 ❯ ~/bin/terraform plan         
local_file.foo: Refreshing state... [id=4bf3e335199107182c6f7638efaad377acc7f452]

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical res
 ```
 
### Steps to Reproduce

Run `terraform plan -destroy` on existing state with terraform `v0.14.X`

### Additional Context

When running `terraform destroy`,  some resources occasionally fail due to cloud resource dependencies. After so many retries, resources are removed from state and I get some `resource not found` errors(AWS launch configs), but when I refresh state prior to trying the destroy, the resource is then found, but fails because it cannot be deleted due to being in use by an autoscaling group. BUT, if I run terraform destroy again, I'm back to the launch configuration not found error, need to refresh state, endless cycle. If `terraform plan -destroy` was actually refreshing state,  this wouldn't happen.

**Comment by jbardin** [comment]
Hi @circa10a,

Thanks for filing the issue. There are a couple cases where things may not have been refreshed automatically during a destroy operation in 0.14, and an explicit `refresh` was required if it prevented a correct plan from being applied.

Since we are no longer developing the 0.14 branch, and 0.15 appears to fix the issue for you, If you have more questions, while we can sometimes help with certain simple problems here, it's better to use [the community forum](https://discuss.hashicorp.com/c/terraform-core) where there are more people ready to help.

Thanks!

**Comment by circa10a** [comment]
@jbardin While 0.15 does address the issue, upgrading for some might not be easily achievable, including myself. We have several thousands of states that need applies with 0.14 before we can upgrade. I don't want to sound like a stickler, but given that 0.14 doesn't behave as expected/documented, I feel that this fix should be backported for those working their way up the terraform upgrade chain considering many are still on 0.12.x. Feel free to tell me to beat it, but I just wanted to express that concern.

**Comment by jbardin** [comment]
Hi @circa10a, I certainly understand and sympathize with the problem here. The changes required to fix the remaining evaluation issues in 0.14 were very extensive and a large part of the 0.15 release. While the changes to accomplish this are too large to make back porting feasible, if they were done what you would essentially be left with is the core of 0.15, so if one can accept that large of a change set, there is little difference in updating to 0.15 itself.

**Comment by circa10a** [comment]
Thanks for the explanation. If anyone comes across this issue, the workaround here is:

```
terraform refresh
terraform plan -destroy
```
